common/autoware_universe_utils/include/autoware/universe_utils/geometry/geometry.hpp

```diff
@@ -583,6 +584,121 @@ std::optional<geometry_msgs::msg::Point> intersect(
  */
 bool intersects_convex(const Polygon2d & convex_polygon1, const Polygon2d & convex_polygon2);
 
+// Alternatives for Boost.Geometry ----------------------------------------------------------------
+
+// TODO(mitukou1109): remove namespace when migrating to alternatives
+namespace alt
+{
+class Vector2d
+{
+public:
+  Vector2d() : x_(0.0), y_(0.0) {}
+
+  Vector2d(const double x, const double y) : x_(x), y_(y) {}
+
+  double cross(const Vector2d & other) const { return x_ * other.y() - y_ * other.x(); }
+
+  double dot(const Vector2d & other) const { return x_ * other.x() + y_ * other.y(); }
+
+  double norm() const { return std::hypot(x_, y_); }
+
+  Vector2d vector_triple(const Vector2d & v1, const Vector2d & v2) const
+  {
+    const auto tmp = this->cross(v1);
+    return {-v2.y() * tmp, v2.x() * tmp};
+  }
+
+  const double & x() const { return x_; }
+
+  double & x() { return x_; }
+
+  const double & y() const { return y_; }
+
+  double & y() { return y_; }
+
+private:
+  double x_;
+  double y_;
+};
+
+using Point2d = Vector2d;
```

Comment:
Why don't you name Point2d to the class directly?
Regarding Eigen, they use vectors and matrices, so Vector2d makes sense. But in this case, since we use the point, the Point2d class sounds better to me.

Reply:
I thought it would be strange if a class representing a point would have functions of vector operations such as inner product and outer product as its members.

Reply:
Makes sense. Thank you for the explanation!
Please write it in this file so that other people understand the reason.